### PR TITLE
[SOL-2101, SOL-2102] Increase text length for dynamic coupons

### PR DESCRIPTION
**Pull request by vkaracic**
The catalog query field needs to be able to store more than 255 characters, therefor it's being changed to a Django TextField field. Additionally a field was added to indicate the length of the query.

https://openedx.atlassian.net/browse/SOL-2101
https://openedx.atlassian.net/browse/SOL-2102

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @vkaracic! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request.

To automatically create an OSPR issue for this pull request, just visit this link: https://openedx-webhooks.herokuapp.com/github/process_pr?number=979&repo=edx%2Fecommerce

**Comment by vkaracic** [comment]
@clintonb do you have time to review this, please?

File: ecommerce/extensions/offer/models.py [hunk 1]
```diff
@@ -75,7 +75,7 @@ class Range(AbstractRange):
         'course_seat_types',
     ]
     catalog = models.ForeignKey('catalogue.Catalog', blank=True, null=True, related_name='ranges')
-    catalog_query = models.CharField(max_length=255, blank=True, null=True)
+    catalog_query = models.TextField(blank=True, null=True)
```
Comment (iivic) [inline ecommerce/extensions/offer/models.py:78]: Should catalog queries have length limit?

File: ecommerce/static/js/views/coupon_form_view.js [hunk 2]
```diff
@@ -279,6 +280,11 @@ define([
                 this._super();
             },
 
+            updateCatalogQueryLength: function() {
+                var query_length = this.$('textarea[name=catalog_query]').val().length;
+                this.$('span.query_length').text(query_length);
```
Comment (iivic) [inline ecommerce/static/js/views/coupon_form_view.js]: Unnecessary long selector why not just `.query_length`?

**Review by iivic — COMMENTED** [review]
Can you provide a ss since ui changed?

**Comment by clintonb** [comment]
Please provide examples of queries that are currently not allowed?

**Review by clintonb — APPROVED** [review]

File: ecommerce/extensions/offer/models.py (hunk 1 above)
Comment (vkaracic) [reply ecommerce/extensions/offer/models.py:78]: I can't find anywhere a limit on haystack queries and it's not indicated on the ticket so I guess not.
@clintonb is there a limit on the query length the course discovery can handle?

Comment (clintonb) [reply ecommerce/extensions/offer/models.py:78]: The only limits are those of URLs/query strings between Otto and the Catalog Service. I haven't found any limits for Python/nginx. The browser limits range anywhere from 2000 to 200000 characters.

**Comment by vkaracic** [comment]
The ticket indicates that queries longer than 255 are a problem. Changing the field to `TextField` should fix this.

**Comment by vkaracic** [comment]
@iivic 
out-1

**Review by iivic — APPROVED** [review]

**Review by mjfrey — APPROVED** [review]

**Pull request merged**